### PR TITLE
feat: Add pos_label directly to EstimatorReport constructor

### PR DESCRIPTION
**Pull request by glemaitre**
Partially addressing #1749 
Partially addressing #1719

This PR allows the following demo to work:

```python
import skore
from skore import EstimatorReport, Project
from sklearn.datasets import fetch_openml
from skrub import tabular_learner

# %%
X, y = fetch_openml("adult", version=2, as_frame=True, return_X_y=True)

# %%
X_train, X_test, y_train, y_test = skore.train_test_split(
    X, y, random_state=1
)

# %%
baseline = tabular_learner('classification')
baseline_report = EstimatorReport(
    baseline,
    X_train=X_train,
    y_train=y_train,
    X_test=X_test,
    y_test=y_test,
    pos_label=">50K"
)
baseline_report.help()

# %%
# create project
project = Project(name="hub://Probabl/test-adult-pos-label")
# %%
project.put("baseline", baseline_report)
```

`pos_label` is passed together with `y` and can overwritten in the metric/plotting methods. But actually remove the need to encode the target if it is not 0/1 or -1/1 in the case of binary classification when we try to serialize the report to send it to the hub.

### CODE REVIEW

File: skore/src/skore/sklearn/_estimator/metrics_accessor.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def report_metrics(
         scoring: Optional[Union[Scoring, list[Scoring]]] = None,
         scoring_names: Optional[Union[ScoringName, list[ScoringName]]] = None,
         scoring_kwargs: Optional[dict[str, Any]] = None,
-        pos_label: Optional[PositiveLabel] = None,
+        pos_label: Optional[PositiveLabel] = "default",
```
Comment (auguste-probabl) [inline skore/src/skore/sklearn/_estimator/metrics_accessor.py:69]: Maybe "auto" instead, to keep things consistent?

Comment (auguste-probabl) [inline skore/src/skore/sklearn/_estimator/metrics_accessor.py:69]:
```suggestion
        pos_label: Union[Optional[PositiveLabel], Literal["default"]] = "default",
```

File: skore/src/skore/sklearn/_estimator/metrics_accessor.py [hunk 2]
```diff
@@ -867,8 +873,9 @@ def recall(
                 only the statistics of the positive class (i.e. equivalent to
                 `average="binary"`).
 
-        pos_label : int, float, bool or str, default=None
-            The positive class.
+        pos_label : int, float, bool, str or None default="default"
```
Comment (auguste-probabl) [inline skore/src/skore/sklearn/_estimator/metrics_accessor.py:876]:
```suggestion
        pos_label : int, float, bool, str or None, default="default"
```

File: skore/src/skore/sklearn/_estimator/report.py [hunk 3]
```diff
@@ -431,6 +439,15 @@ def y_test(self, value):
         self._y_test = value
         self._initialize_state()
 
+    @property
+    def pos_label(self) -> Optional[PositiveLabel]:
```
Comment (auguste-probabl) [inline skore/src/skore/sklearn/_estimator/report.py:443]:
```suggestion
    def pos_label(self) -> Union[PositiveLabel, None]:
```

**Review by auguste-probabl — CHANGES_REQUESTED** [review]
Mostly nitpicks

File: skore/src/skore/sklearn/_estimator/metrics_accessor.py (hunk 1 above)
Comment (glemaitre) [reply skore/src/skore/sklearn/_estimator/metrics_accessor.py:69]: The pattern that I already saw would be to use a sentinel:

```python
_SENTINEL = object()

def func(param=_SENTINEL):
    pass
    
```

File: skore/src/skore/sklearn/_estimator/metrics_accessor.py [hunk 4]
```diff
@@ -867,8 +873,9 @@ def recall(
...
+            If `"default"`, the positive class is set to the one provided when creating
+            the report. Use this parameter to override the positive class.
```
[the remark this answers is not in the document]
Comment (glemaitre) [reply skore/src/skore/sklearn/_estimator/metrics_accessor.py]: Good remark.

**Comment by glemaitre** [comment]
@auguste-probabl it should be ready to be reviewed.

File: skore/tests/unit/sklearn/estimator/test_estimator.py [hunk 5]
```diff
@@ -442,6 +460,31 @@ def test_estimator_report_display_binary_classification(
...
+        getattr(report.metrics, display)()
+
+    report = EstimatorReport(classifier, X_test=X, y_test=y, pos_label="A")
+    disp = getattr(report.metrics, display)()
+    disp.plot()
+    assert "Positive label: A" in disp.ax_.get_xlabel()
+
+    disp = getattr(report.metrics, display)(pos_label="B")
+    disp.plot()
+    assert "Positive label: B" in disp.ax_.get_xlabel()
```
Comment (auguste-probabl) [inline skore/tests/unit/sklearn/estimator/test_estimator.py:485]:
```suggestion
@pytest.mark.parametrize("metric", ["roc", "precision_recall"])
def test_estimator_report_display_binary_classification_pos_label(
    pyplot, binary_classification_data, display
):
    """Check the behaviour of the display methods when `pos_label` needs to be set."""
    X, y = make_classification(
        n_classes=2, class_sep=0.8, weights=[0.4, 0.6], random_state=0
    )
    labels = np.array(["A", "B"], dtype=object)
    y = labels[y]
    classifier = LogisticRegression().fit(X, y)
    report = EstimatorReport(classifier, X_test=X, y_test=y)
    with pytest.raises(ValueError, match="pos_label is not specified"):
        getattr(report.metrics, metric)()

    report = EstimatorReport(classifier, X_test=X, y_test=y, pos_label="A")
    display = getattr(report.metrics, metric)()
    display.plot()
    assert "Positive label: A" in display.ax_.get_xlabel()

    display = getattr(report.metrics, metric)(pos_label="B")
    display.plot()
    assert "Positive label: B" in display.ax_.get_xlabel()
```

**Review by auguste-probabl — APPROVED** [review]
Alright, good to go!
